python/sglang/test/test_flashinfer_cutedsl_scaled_mm.py

```diff
@@ -0,0 +1,419 @@
+import pytest
+import torch
+from flashinfer import fp4_quantize
+from sgl_kernel import scaled_fp4_grouped_quant
+from torch.nn import functional as F
+
+from sglang.srt.layers.moe.flashinfer_cutedsl_moe import flashinfer_cutedsl_moe_masked
+
+skip_condition = torch.cuda.get_device_capability() < (10, 0)
+
+FLOAT8_E4M3_MAX = 448.0
+FLOAT4_E2M1_MAX = 6.0
+
+SHAPES = [(2, 128, 256, 8, 2), (16, 128, 512, 8, 4)]
+
+
+def convert_swizzled_to_linear(a_sf_swizzled: torch.Tensor, m, k, block_size):
+    m_tiles = (m + 128 - 1) // 128
+    f = block_size * 4
+    k_tiles = (k + f - 1) // f
+    tmp = torch.reshape(a_sf_swizzled, (1, m_tiles, k_tiles, 32, 4, 4))
+    tmp = torch.permute(tmp, (0, 1, 4, 3, 2, 5))
+    out = tmp.reshape(m_tiles * 128, k_tiles * f // block_size)
+    return out[0:m, 0:k]
+
+
+def break_fp4_bytes(a, dtype):
```

Comment:
can we use the exsting convert_swizzled_to_linear and break_fp4_bytes from `test_fp4_moe.py`? or put them into test_utils.py?

Reply:
Or, since this is mainly about fp4. Maybe it also reasonable to put the test into test_fp4_moe.py?